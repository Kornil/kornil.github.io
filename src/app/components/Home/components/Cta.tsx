import React from "react";

import OwnerContext from "app/shared/OwnerContext";
import StyledCta from "./StyledCta";

const Cta: React.SFC<any> = props => (
  <StyledCta.Container>
    <StyledCta.Header>{props.OwnerContext.name}</StyledCta.Header>
    <p>
      <em>I breathe JavaScript</em>
    </p>
  </StyledCta.Container>
);

export default OwnerContext.connect(Cta);
