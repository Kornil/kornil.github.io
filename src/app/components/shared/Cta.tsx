import React from "react";

import OwnerContext from "app/shared/OwnerContext";

const Cta = (props: any) => <h1>{props.OwnerContext.name}</h1>;

export default OwnerContext.withContext(Cta);
