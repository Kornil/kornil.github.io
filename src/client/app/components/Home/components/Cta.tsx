import React from "react";

import OwnerContext, {
  OwnerContextValueInterface
} from "app/context/OwnerContext";
import StyledCta from "./StyledCta";

const Cta: React.SFC<{ OwnerContext: OwnerContextValueInterface }> = props => (
  <StyledCta.Container>
    <StyledCta.Header>{props.OwnerContext.name}</StyledCta.Header>
    <p>
      <em>I breathe JavaScript</em>
    </p>
  </StyledCta.Container>
);

export default OwnerContext.connect(Cta);
