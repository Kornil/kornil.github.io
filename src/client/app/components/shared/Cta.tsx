import React from "react";
import styled, { StyledComponent } from "react-emotion";

interface StyledCtaInterface {
  [key: string]: StyledComponent<{}, any, any>;
}

const StyledCta: StyledCtaInterface = {
  Container: styled.section`
    align-self: center;
    justify-self: center;
  `,
  Header: styled.h1`
    margin-top: 0;
  `
};

const Cta: React.SFC<{ title: string; body: React.ReactNode }> = ({
  title,
  body
}) => (
  <StyledCta.Container>
    <StyledCta.Header>{title}</StyledCta.Header>
    {body}
  </StyledCta.Container>
);

export default Cta;
