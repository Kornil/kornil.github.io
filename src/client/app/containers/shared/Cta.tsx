import React from "react";
import styled, { StyledComponent } from "react-emotion";

interface StyledCtaInterface {
  [key: string]: StyledComponent<{}, any, any>;
}

const StyledCta: StyledCtaInterface = {
  Container: styled.section`
    align-self: center;
    justify-self: center;
  `,
  Header: styled.h1`
    margin-top: 0;
  `
};

const Cta: React.SFC<{
  title: string | React.ReactNode;
  children: React.ReactNode;
}> = ({ title, children }) => (
  <StyledCta.Container>
    <StyledCta.Header>{title}</StyledCta.Header>
    {children}
  </StyledCta.Container>
);

export default Cta;
