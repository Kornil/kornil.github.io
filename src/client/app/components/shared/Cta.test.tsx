import { shallow } from "enzyme";
import React from "react";

import Cta from "./Cta";

test("it renders correctly", () => {
  const wrapper = shallow(
    <Cta title="hello">
      <p>world</p>
    </Cta>
  );
  expect(wrapper).toMatchSnapshot();
});
