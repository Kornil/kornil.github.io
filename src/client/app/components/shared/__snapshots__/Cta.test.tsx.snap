// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<Container>
  <Header>
    hello
  </Header>
  <p>
    world
  </p>
</Container>
`;
